de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/generator/OvertargetGenerator.xtend

```diff
@@ -28,10 +31,21 @@ import de.dlr.sc.overtarget.language.targetmodel.TargetmodelPackage
  */
 class OvertargetGenerator extends AbstractGenerator {
 	static val DEFAULT_JRE_CONTAINER = "org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/";
-	
+
 	override void doGenerate(Resource resource, IFileSystemAccess2 fsa, IGeneratorContext context) {
 		for (model : resource.allContents.toIterable.filter(TargetModel)) {
-			fsa.generateFile(model.name + ".target", OvertargetOutputConfigurationProvider.GENERATOR_OUTPUT_ID_OVERTARGET, model.compile)
+			EcoreUtil.resolveAll(resource)
+			if (ReferenceTargetHelper.importedModelIsProxy(model) == true || ReferenceTargetHelper.parentIsProxy(model) == true) {
+				ReferenceTargetHelper.getModelToGenerate(model)
```

Comment:
And can we maybe move that code to a new method... Just to make it better readable? To a method that is called something like "generateTargetForReferenced()" or so... maybe with a better name :D

Reply:
I just called it generateReferencedTarget(), is that fine?

Reply:
Well your not really generating the referencedTarget, are you? The ReferencedTarget is the be the tmodel which your reference targets... We're not generating this one but we're generating a new tmodel that helps us resolve the referenced tmodel ;)

Reply:
yeah, you're right!